File: fibonacci/fibonacci.pl
#!/usr/bin/env perl

use strict;
use warnings;
use bigint;

sub fibonacci {
    my ($n) = @_;
    my $a = 0;
    my $b = 1;
    
    for (my $i = 0; $i < $n; $i++) {
        my $temp = $a;
        $a = $b;
        $b += $temp;
    }
    
    return $a;
}

my $result = fibonacci(100000);
print "$result\n";

File: fibonacci/fibonacci.cs
using System;

public static class Fibonacci
{
    public static ulong Calculate(int n)
    {
        ulong a = 0, b = 1;
        for (int i = 0; i < n; i++)
        {
            ulong temp = a;
            a = b;
            b += temp;
        }
        return a;
    }
    
    public static void Main()
    {
        ulong result = Calculate(100000);
        Console.WriteLine(result);
    }
}
